libs/entitlement/src/main/java/org/elasticsearch/entitlement/package-info.java

```diff
@@ -0,0 +1,189 @@
+ *
+ * <p>
+ * {@link org.elasticsearch.entitlement.initialization.EntitlementInitialization} builds the set of methods to instrument using
+ * {@link org.elasticsearch.entitlement.instrumentation.InstrumentationService#lookupMethods} on the version-specific subclass of
+ * {@link org.elasticsearch.entitlement.bridge.EntitlementChecker}.
+ * {@link org.elasticsearch.entitlement.bridge.EntitlementChecker} is the interface that contains the definition of all the check methods;
+ * it needs to be accessible by both this project and the code injected by the agent, therefore is located in a small, self-contained
+ * library ({@see the {@code bridge} subproject}).
+ * </p>
+ * <p>
+ * See {@link org.elasticsearch.entitlement.instrumentation.InstrumentationService#lookupMethods} for details.
+ * </p>
+ *
+ * <h3>How that works across different Java versions</h3>
+ * <p>
+ * The {@code bridge} subproject uses multi-release jars via the {@code mrjar} plugin, which makes it is possible to specify classes for
+ * specific Java versions in specific {@code src} folders (e.g. {@code main23} for classes available to Java 23+).
+ * </p>
+ * <p>
+ * See the {@code bridge} subproject for details.
+ * </p>
+ *
+ * <h2>Prologue injection</h2>
+ *
+ * <p>
+ * Agents get access to the Java instrumentation API by receiving a {@link java.lang.instrument.Instrumentation} instance, which we pass
+ * to {@link org.elasticsearch.entitlement.initialization.EntitlementInitialization#initialize} to setup code needed to transform classes
+ * as they get loaded. See {@link org.elasticsearch.entitlement.initialization.EntitlementInitialization#initialize} for details.
+ * </p>
+ * <p>
+ * Our implementation instrument classes by adding a prologue to the methods identified in
+ * {@link org.elasticsearch.entitlement.initialization.EntitlementInitialization} (see previous section).
+ * See {@link org.elasticsearch.entitlement.instrumentation.Instrumenter#instrumentClass} for details.
+ * </p>
+ *
+ * <h2>Caller identification</h2>
+ *
+ * <p>
+ * In order to verify if a method is entitled to perform an action, we need to identify the right policy to check; the first step here is
+ * to identify the caller. This is done in the injected prologue, via a helper function
+ * {@link org.elasticsearch.entitlement.bridge.Util#getCallerClass}, which performs a limited stack walk (it "walks up" 2 frames) to
+ * identify the caller class. This utility is similar to {@link java.lang.StackWalker#getCallerClass}, but handles the case in which it
+ * is called from the "outermost frame", e.g. when a method is called from a native frame.
+ * </p>
+ * <p>
+ * Our plan here was to avoid the stack walking entirely for the most cases, by using the {@code CallerSensitive} attribute together with
+ * the {@code Reflection.getCallerClass()} method; the current implementation (8.18/9.0) does not include this optimization, but we plan to
+ * introduce that in a later release.
```

Comment:
I'm not sure these thoughts on future plans belong in the javadocs?

Reply:
Removed